src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -1972,16 +2020,22 @@ void CameraAVStreamMgmtServer::HandleSnapshotStreamAllocate(HandlerContext & ctx
 
     // Check if the streamID matches an existing one in the
     // mAllocatedSnapshotStreams.
-    bool streamExists = std::find_if(mAllocatedSnapshotStreams.begin(), mAllocatedSnapshotStreams.end(),
-                                     [&snapshotStreamID](const SnapshotStreamStruct & entry) {
-                                         return entry.snapshotStreamID == snapshotStreamID;
-                                     }) != mAllocatedSnapshotStreams.end();
-    if (!streamExists)
+    auto it = std::find_if(
+        mAllocatedSnapshotStreams.begin(), mAllocatedSnapshotStreams.end(),
+        [snapshotStreamID](const SnapshotStreamStruct & sStream) { return sStream.snapshotStreamID == snapshotStreamID; });
+
+    if (it == mAllocatedSnapshotStreams.end())
     {
         // Add the allocated snapshotstream object in the AllocatedSnapshotStreams list.
         snapshotStreamArgs.snapshotStreamID = snapshotStreamID;
         AddSnapshotStream(snapshotStreamArgs);
     }
+    else
+    {
+        SnapshotStreamStruct & sStreamToUpdate = *it;
+        // Reusing the existing stream. Update range parameters
+        UpdateSnapshotStreamRangeParams(sStreamToUpdate, snapshotStreamArgs);
```

Comment:
rename snapshotStreamArgs to snapshotStream to align with snapshotStreamToUpdate